apps/opik-backend/src/main/java/com/comet/opik/infrastructure/llm/gemini/LlmProviderGemini.java

```diff
@@ -40,8 +45,38 @@ public void validateRequest(@NonNull ChatCompletionRequest request) {
     }
 
     @Override
-    public Optional<ErrorMessage> getLlmProviderError(@NonNull Throwable runtimeException) {
+    public Optional<ErrorMessage> getLlmProviderError(@NonNull Throwable throwable) {
+        /// gemini throws RuntimeExceptions with message structure as follows:
+        /// ```
+        /// java.lang.RuntimeException: HTTP error (429): {
+        ///   "error": {
+        ///     "code": 429,
+        ///     "message": "Resource has been exhausted (e.g. check quota).",
+        ///     "status": "RESOURCE_EXHAUSTED"
+        ///   }
+        /// }
+        ///  ```
+        String message = throwable.getMessage();
+        if (message.contains("{")) {
+            String jsonPart = message.substring(message.indexOf("{")); // Extract JSON part
+            try {
+                // Parse JSON
+                JsonNode errorNode = JsonUtils.MAPPER.readTree(jsonPart).get("error");
+                if (errorNode != null) {
+                    var code = errorNode.get("code").asInt();
+                    if (familyOf(code) == Response.Status.Family.CLIENT_ERROR) {
```

Comment:
If we are familiar with the structure they send use, they not creating a DTO and trying to parse the DTO?

Reply:
Done